utils/hct/gen_intrin_main.txt

```diff
@@ -372,6 +372,7 @@ resource [[hidden]] CreateResourceFromHeap(in uint index);
 $match<1, 0> bool<> [[rn]] and(in any<> x, in $type1 y);
 $match<1, 0> bool<> [[rn]] or(in any<> x, in $type1 y);
 $type2 [[rn]] select(in bool<> cond, in $match<1, 2> any<> t, in $type2 f);
+$type2 [[rn]] select(in bool cond, in any_sampler t, in $type2 f);
```

Comment:
Am I correct in assuming this is required because `any` isn't really _anything_ but rather any scalar or vector?
Do we need any other overloads for this? Have we tested if it works with textures?

Reply:
You are correct that `any` isn't really "any". It's just scalars and vectors. I had an earlier change that created a broader type. However, since all that's needed here are samplers since textures didn't work with ternary operators before so they aren't expected to work with `sample()`, I chose to simplify here. We might consider allowing textures for the sake of consistency, but that seems like a separate change rather than something that is needed to make HLSL 2021 consistent with what came before.